targets/TARGET_ONSEMI/TARGET_NCS36510/gpio_irq_api.c

```diff
@@ -193,40 +200,40 @@ void gpio_irq_free(gpio_irq_t *obj)
  */
 void gpio_irq_set(gpio_irq_t *obj, gpio_irq_event event, uint32_t enable)
 {
-
     /* Enable the GPIO clock */
-    CLOCK_ENABLE(CLOCK_GPIO);
+    if(!CLOCK_IS_ENABLED(CLOCK_GPIO))
+    {
+        CLOCK_ENABLE(CLOCK_GPIO);
+    }
 
     switch(event) {
-        case IRQ_RISE:
-            obj->GPIOMEMBASE->IRQ_EDGE = (obj->pinMask);
-            obj->GPIOMEMBASE->IRQ_LEVEL = (IO_ALL ^ (obj->pinMask));
-            /* Enable is an integer; hence checking for 1 or 0*/
-            if (enable == 1) {
-                /* Enable rising edge */
-                obj->GPIOMEMBASE->IRQ_POLARITY_SET = (obj->pinMask);
-            } else if (enable == 0) {
-                /* Disable rising edge */
-                obj->GPIOMEMBASE->IRQ_POLARITY_SET = (IO_ALL ^ (obj->pinMask));
-            }
-            break;
-
-        case IRQ_FALL:
-            obj->GPIOMEMBASE->IRQ_EDGE = (obj->pinMask);
-            obj->GPIOMEMBASE->IRQ_LEVEL = (IO_ALL ^ (obj->pinMask));
-            /* Enable is an integer; hence checking for 1 or 0*/
-            if (enable == 1) {
-                /* Enable falling edge */
-                obj->GPIOMEMBASE->IRQ_POLARITY_CLEAR = (obj->pinMask);
-            } else if (enable == 0) {
-                /* Disable falling edge */
-                obj->GPIOMEMBASE->IRQ_POLARITY_CLEAR = (IO_ALL ^ (obj->pinMask));
-            }
-            break;
+    case IRQ_RISE:
+        obj->GPIOMEMBASE->IRQ_EDGE |= obj->pinMask;
+        /* Enable is an integer; hence checking for 1 or 0*/
+        if (enable == 1) {
+            /* Enable rising edge */
+            obj->GPIOMEMBASE->IRQ_POLARITY_SET |= obj->pinMask;
+        } else if (enable == 0) {
+            /* Disable rising edge */
+            obj->GPIOMEMBASE->IRQ_POLARITY_SET |= (IO_ALL ^ (obj->pinMask));
```

Comment:
enable 0 doesn't mean "disable rising edge" - it's a request to disable the interrupts.

I'd suggest two separate tests - one to program IRQ_POLARITY_SET or CLEAR depending on event, and a separate one to program IRQ_ENABLE_SET or IRQ_ENABLE_CLEAR depending on event.

And again, all of those just being `= obj->pinMask`, if my register map is correct. (And it seems logical to me).